modules/cxf/src/main/java/org/mule/module/cxf/MuleInvoker.java

```diff
@@ -125,11 +126,17 @@ else if(e instanceof ComponentException) {
             responseEvent = null;
         }
 
+        if( responseEvent instanceof NonBlockingVoidMuleEvent)
+        {
+            exchange.put(CxfConstants.MULE_EVENT, responseEvent);
+            return new Object();
```

Comment:
Yes, just something different from null is required here as output value, due to the important value is stored in the exchange.